core/src/main/java/org/mobilitydata/gtfsvalidator/validator/ValidatorProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import java.util.List;
+import org.mobilitydata.gtfsvalidator.table.GtfsEntity;
+import org.mobilitydata.gtfsvalidator.table.GtfsFeedContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTableContainer;
+
+/**
+ * Provider of all kinds of validators for fields, entities and files.
+ *
+ * <p>A {@code ValidatorProvider} is a handy pack of validators passed to different parts of the
+ * system. Unit tests and users may provide their own implementations of {@code ValidatorProvider}.
+ */
+public interface ValidatorProvider {
```

Comment:
You see, clients may inherit from DefaultValidatorProvider and override only some methods.

ValidatorProvider as an interface may be fully replaced in a unit tests that does not want to perform any extra validation.

In general, my feeling is that having an interface is cleaner.

Reply:
>clients may inherit from DefaultValidatorProvider and override only some methods.

This is nice! Also could be used in the future to help support profiles bundled with the canonical validator as well, if I understand correctly. It would be nice to have some documentation in a Markdown file for how this works and example use.